feat: add global shortcuts to togglePlay, playNext and playPrevious

daiyam opened the pull request:
This PR adds global shortcuts to media shortcuts.

daiyam commented:
On macOS, the app crashs if it's not authorize to have global shortcuts (in `Privacy & Security`/`Accessibility`)...

basharovV commented:
I checked this and F8 play/pause works but F7 and F9 don't work as previous/next keys. What device/OS do you have?

To do this without hijacking the keys from other apps or requesting Accessibility permissions the app needs to somehow gain Now Playing status so it can be controlled via the media keys like other media apps. Docs: [Becoming a Now Playable app](https://developer.apple.com/documentation/mediaplayer/becoming_a_now_playable_app?language=objc). 

So I would continue the work in the `mediakeys` module - need to find out why macOS is not treating Musicat as a Now Playable app, even though it looks like it meets the eligibility requirements: a) displays the track info in the Now Playing center, and b) registers a remote command center action (currently just for play/pause)

daiyam commented:
macos but with a logitech keyboard.

daiyam commented:
I've added some logics to avoid the crash at the start when it was missing the permissions.

I've also added some `global_shortcut.unregister` to try to make it work on your mac.
